### PR TITLE
Allow deny listing balancer pools

### PR DESCRIPTION
**Pull request by vkgnosis**
Fixes https://github.com/cowprotocol/services/issues/242 .

### Test Plan

Used the new manual pool fetching test to see that this pool no longer gets returned

### CODE REVIEW

File: crates/shared/src/sources/balancer_v2/pool_fetching.rs [hunk 1]
```diff
@@ -370,6 +384,54 @@ mod tests {
...
+                web3: web3.clone(),
+            })));
+        let block_stream =
+            crate::current_block::current_block_stream(web3.clone(), Duration::from_secs(1000))
+                .await
+                .unwrap();
+        let deny_list = vec![H256(hex_literal::hex!(
+            "072f14b85add63488ddad88f855fda4a99d6ac9b000200000000000000000027"
+        ))];
+        // let deny_list = vec![];
```
Comment (sunce86) [inline crates/shared/src/sources/balancer_v2/pool_fetching.rs:405]: nit: probably leftover

**Review by sunce86 — APPROVED** [review]
lgtm

**Review by josojo — APPROVED** [review]
lgtm

File: crates/shared/src/sources/balancer_v2/pool_fetching.rs (hunk 1 above)
Comment (vkgnosis) [reply crates/shared/src/sources/balancer_v2/pool_fetching.rs:405]: That was intentional so I could compare the output with and without deny list. Makes it easier for future testers so I'd leave it in.

**Pull request merged**